core/src/test/scala/com.codacy.analysis.core/git/RepositorySpec.scala

```diff
@@ -35,5 +37,68 @@ class RepositorySpec extends Specification with NoLanguageFeatures {
       }
     }
 
+    "get all uncommited changes" in {
+
+      "changed files" in {
+        withTemporaryGitRepo { directory =>
+          (for {
+            file <- File.temporaryFile(parent = Some(directory))
+          } yield {
+
+            Process(Seq("git", "add", "."), directory.toJava).!
+            Process(Seq("git", "commit", "-m", "geez rick! i-i-i-i-i a-a-a-added a file!"), directory.toJava).!
+
+            file.write("Wubba lubba dub dub!")
+
+            Git.repository(directory).flatMap(_.uncommitedFiles) must beLike {
+              case Success(uncommited) =>
+                uncommited.foreach(println)
+                uncommited must contain(relativePath(file, directory))
+            }
+          }).get
+        }
+      }
+
+      "untracked files" in {
+        "with an untracked folder that contains an untracked file" in {
+          withTemporaryGitRepo { directory =>
+            (for {
+              mainFolder1 <- File.temporaryDirectory(parent = Some(directory))
+              subFolder <- File.temporaryDirectory(parent = Some(mainFolder1))
+              deepFile <- File.temporaryFile(parent = Some(subFolder))
+            } yield {
+              Git.repository(directory).flatMap(_.uncommitedFiles) must beLike {
+                case Success(uncommited) =>
+                  uncommited must contain(relativePath(deepFile, directory))
+              }
+            }).get
+          }
+        }
+
+        "with an untracked folder with no content" in {
+          withTemporaryGitRepo { directory =>
+            (for {
+              file1 <- File.temporaryFile(parent = Some(directory))
+              file2 <- File.temporaryFile(parent = Some(directory))
+              mainFolder1 <- File.temporaryDirectory(parent = Some(directory))
+              subFolder <- File.temporaryDirectory(parent = Some(mainFolder1))
+              deepFile <- File.temporaryFile(parent = Some(subFolder))
+              mainFolder2 <- File.temporaryDirectory(parent = Some(directory))
+              noContentsFolder <- File.temporaryDirectory(parent = Some(mainFolder2))
+            } yield {
+              Git.repository(directory).flatMap(_.uncommitedFiles) must beLike {
+                case Success(uncommited) =>
+                  uncommited must not contain relativePath(noContentsFolder, directory)
+              }
+            }).get
+          }
+        }
+      }
+    }
+
+  }
+
+  private def relativePath(targetFile: File, directory: File): String = {
+    targetFile.pathAsString.replace(s"${directory.pathAsString}/", "")
```

Comment:
Better files `.relativize` should do the trick

Reply:
I really hate relativize since it can destroy paths. I actually prefer the strip prefix since it is safer.